earth2studio/models/dx/corrdiff.py

```diff
@@ -45,10 +51,672 @@
+            raise ValueError("`number_of_steps` must be a positive integer.")
+        if solver not in ["heun", "euler"]:
+            raise ValueError(f"{solver} is not supported, must be in ['heun', 'euler']")
+
+        # Store model configuration
+        self.number_of_samples = number_of_samples
+        self.number_of_steps = number_of_steps
+        self.solver = solver
+        self.inference_mode = inference_mode
+        self.hr_mean_conditioning = hr_mean_conditioning
+        self.seed = seed
+        self.img_shape = (lat_output_grid.shape[0], lon_output_grid.shape[0])
+
+        # Store models
+        self.residual_model = residual_model
+        self.regression_model = regression_model
+
+        # Store variable names
+        self.input_variables = input_variables
+        self.output_variables = output_variables
+
+        # Register buffers for model parameters
+        self._register_buffers(
+            lat_input_grid,
+            lon_input_grid,
+            lat_output_grid,
+            lon_output_grid,
+            in_center,
+            in_scale,
+            invariant_center,
+            invariant_scale,
+            out_center,
+            out_scale,
+            invariants,
+        )
+
+        # Set up sampler
+        self.sampler = self._setup_sampler(sampler_type)
+
+    def _check_latlon_grid(
+        self,
+        lat_input_grid: torch.Tensor,
+        lon_input_grid: torch.Tensor,
+        lat_output_grid: torch.Tensor,
+        lon_output_grid: torch.Tensor,
+    ) -> None:
+        """
+        Perform the following checks on the passed lat/lon grids.
+
+            - Check if the lat/lon grids are regular and have the same resolution.
+            - Check that the output lat/lon grids are a subset of the input lat/lon grids.
+
+        """
+        input_grid_n_dim = len(lat_input_grid.shape)
+
+        if input_grid_n_dim == 1:
+            lat_input_grid = lat_input_grid.unsqueeze(1)
+            lon_input_grid = lon_input_grid.unsqueeze(0)
+
+        if not torch.allclose(
+            lat_input_grid[1, 0] - lat_input_grid[0, 0],
+            lon_input_grid[0, 1] - lon_input_grid[0, 0],
+            rtol=1e-3,
+            atol=1e-3,
+        ):
+            raise ValueError(
+                f"Input lat/lon grids must have the same resolution. Received lat_input_grid[1,0] - lat_input_grid[0,0] = {lat_input_grid[1,0] - lat_input_grid[0,0]} and lon_input_grid[0,1] - lon_input_grid[0,0] = {lon_input_grid[0,1] - lon_input_grid[0,0]}"
+            )
+        if not torch.all(lat_output_grid >= lat_input_grid.min()) or not torch.all(
+            lat_output_grid <= lat_input_grid.max()
+        ):
+            raise ValueError(
+                f"Output lat/lon grids must be a subset of the input lat/lon grids. Received lat_output_grid.min() = {lat_output_grid.min()} and lat_output_grid.max() = {lat_output_grid.max()}"
+            )
```

Comment:
@dallasfoster Minor comment, do we want the output grid to be a subset of the input grid? I mean, if the input resolution is much lower one input pixel could cover quite a large area of the output grid, so the output grid would still fall inside the input grid even if the min / max coordinates are smaller / bigger